File: core/data/tests/can_generate_generic_struct/output.scala
package com.agilebits

package onepassword {

case class GenericStruct[A, B] (
	field_a: A,
	field_b: Vector[B]
)

case class UnusedGenericsStruct (
	field_a: Float,
	field_b: Float
)

case class UnusedGenericsEmptyStruct ()

case class GenericStructUsingGenericStruct[T] (
	struct_field: GenericStruct[String, T],
	second_struct_field: GenericStruct[T, String],
	third_struct_field: GenericStruct[T, Vector[T]]
)

sealed trait EnumUsingGenericStruct {
	def serialName: String
}
object EnumUsingGenericStruct {
	case class VariantA(content: GenericStruct[String, Float]) extends EnumUsingGenericStruct {
		val serialName: String = "VariantA"
	}
	case class VariantB(content: GenericStruct[String, Int]) extends EnumUsingGenericStruct {
		val serialName: String = "VariantB"
	}
	case class VariantC(content: GenericStruct[String, Boolean]) extends EnumUsingGenericStruct {
		val serialName: String = "VariantC"
	}
	case class VariantD(content: GenericStructUsingGenericStruct[Unit]) extends EnumUsingGenericStruct {
		val serialName: String = "VariantD"
	}
}

}

File: core/data/tests/can_generate_generic_struct/output.ts
export interface GenericStruct<A, B> {
	field_a: A;
	field_b: B[];
}

export interface UnusedGenericsStruct {
	field_a: number;
	field_b: number;
}

export interface UnusedGenericsEmptyStruct {
}

export interface GenericStructUsingGenericStruct<T> {
	struct_field: GenericStruct<string, T>;
	second_struct_field: GenericStruct<T, string>;
	third_struct_field: GenericStruct<T, T[]>;
}

export type EnumUsingGenericStruct = 
	| { type: "VariantA", content: GenericStruct<string, number> }
	| { type: "VariantB", content: GenericStruct<string, number> }
	| { type: "VariantC", content: GenericStruct<string, boolean> }
	| { type: "VariantD", content: GenericStructUsingGenericStruct<undefined> };

